### PR TITLE
MAT-613: Support startExitFor[Metadata]MintableBurntToken + minor refactoring

### PR DESCRIPTION
**Pull request by atvanguard**
- Deleted artifacts/ - Importing them from matic-protocol package.
- Removed src/helpers/ - Importing them from matic-protocol package.
- Wrapper over MintableERC721Predicate.startExitForMetadataMintableBurntToken.
- Test is [here](https://github.com/maticnetwork/contracts/pull/154/files#diff-8c611ff46bd52e99ad13dc05846a705a).

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -159,28 +159,41 @@ export default class Matic extends ContractsBase {
...
     return this.withdrawManager.burnERC721Token(token, tokenId, options)
   }
 
   withdraw(txHash: string, options?: SendOptions) {
     return this.withdrawManager.startExitWithBurntERC20Tokens(txHash, options)
   }
 
-  confirmWithdrawERC721(txHash: string, options?: SendOptions) {
+  withdrawNFT(txHash: string, options?: SendOptions) {
+    if (options && !options.from) {
```
Comment (ptsayli) [inline src/index.ts]: add `txHash` exists validation

File: src/root/WithdrawManager.ts [hunk 2]
```diff
@@ -169,61 +172,51 @@ export default class WithdrawManager extends ContractsBase {
...
-      ),
-      options,
-    )
+  /**
+   * Start an exit for a token with metadata (token uri) that was minted and burnt on the side chain
+   * Wrapper over contract call: [MintableERC721Predicate.startExitForMintableBurntToken](https://github.com/maticnetwork/contracts/blob/e2cb462b8487921463090b0bdcdd7433db14252b/contracts/root/predicates/MintableERC721Predicate.sol#L31)
+   * @param burnTxHash Hash of the burn transaction on Matic
+   * @param predicate address of MintableERC721Predicate
+   */
+  async startExitForMintableBurntToken(burnTxHash, predicate: address, options?) {
```
Comment (ptsayli) [inline src/root/WithdrawManager.ts]: any specific reason to pass `predicate address` as a parameter,
can't we directly use from global variable `erc721Predicate: Contract`

File: src/root/WithdrawManager.ts [hunk 3]
```diff
@@ -169,61 +172,51 @@ export default class WithdrawManager extends ContractsBase {
...
-      ),
-      options,
-    )
+  /**
+   * Start an exit for a token with metadata (token uri) that was minted and burnt on the side chain
+   * Wrapper over contract call: [MintableERC721Predicate.startExitForMetadataMintableBurntToken](https://github.com/maticnetwork/contracts/blob/e2cb462b8487921463090b0bdcdd7433db14252b/contracts/root/predicates/MintableERC721Predicate.sol#L66)
+   * @param burnTxHash Hash of the burn transaction on Matic
+   * @param predicate address of MintableERC721Predicate
+   */
+  async startExitForMetadataMintableBurntToken(burnTxHash, predicate: address, options?) {
```
Comment (ptsayli) [inline src/root/WithdrawManager.ts]: same

File: src/root/WithdrawManager.ts (hunk 2 above)
Comment (atvanguard) [reply src/root/WithdrawManager.ts]: Discussed offline.

File: src/index.ts (hunk 1 above)
Comment (atvanguard) [reply src/index.ts]: Done.

**Pull request merged**